packages/vite/src/node/plugins/worker.ts

```diff
@@ -42,61 +71,100 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
           code: `import '${ENV_PUBLIC_PATH}'\n` + _
         }
       }
-      if (
-        query == null ||
-        (query && (query.worker ?? query.sharedworker) == null)
-      ) {
+
+      const workerConstructor = buildWorkerConstructor(query)
+      if (query == null || workerConstructor == null) {
         return
       }
 
       let url: string
       if (isBuild) {
-        if (query.inline != null) {
-          // bundle the file as entry to support imports and inline as blob
-          // data url
-          const rollup = require('rollup') as typeof Rollup
-          const bundle = await rollup.rollup({
-            input: cleanUrl(id),
-            plugins: config.plugins as Plugin[]
+        // bundle an inline script version of the worker
+        const { rollup } = require('rollup') as typeof Rollup
+        const bundle = await rollup({
+          input: cleanUrl(id),
+          plugins: config.plugins as Plugin[]
```

Comment:
This will break current builds (that might currently work in chrome) because we are now running all worker imports through rollup a second time with an identical set of plugins which confuses the plugins because buildStart is called twice with identical parameters. Switching this to `await resolvePlugins({ ...config }, [], [], [])` should fix that (although that will build worker bundles with built in plugins, and not user config added plugins)